src/simplewallet/simplewallet.cpp

```diff
@@ -5070,6 +5064,67 @@ bool simple_wallet::stake(const std::vector<std::string> &args_)
     return true;
   }
 
+  try
+  {
+    const auto& response = m_wallet->get_service_node_list();
+    bool found = false;
+    bool full = false;
+    uint64_t can_contrib_total = 0;
+    uint64_t must_contrib_total = 0;
+    for (const auto& snode_info : response.service_node_states)
+      if (string_to_public_key(snode_info.service_node_pubkey) == service_node_key)
+      {
+        found = true;
+        if (snode_info.contributors.size() < MAX_NUMBER_OF_CONTRIBUTORS)
+          can_contrib_total = snode_info.staking_requirement - snode_info.total_reserved;
+        else
+          full = true;
+        for (const auto& contributor : snode_info.contributors)
+          if (string_to_address(contributor.address) == address)
+          {
+            uint64_t max_increase_reserve = snode_info.staking_requirement - snode_info.total_reserved;
+            uint64_t max_increase_amount_to = contributor.reserved + max_increase_reserve;
+            can_contrib_total = max_increase_amount_to - contributor.amount;
+            must_contrib_total = contributor.reserved - contributor.amount;
+            full = false;
+          }
+      }
+    if (!found)
+    {
+      fail_msg_writer() << tr("Could not find service node in service node list, please make sure it is registered first.");
+      return true;
+    }
+    if (full)
+    {
+      fail_msg_writer() << tr("This service node already has the maximum number of participants, and the specified address is not one of them");
+      return true;
+    }
+    if (can_contrib_total == 0)
+    {
+      fail_msg_writer() << tr("You may not contribute any more to this service node");
+      return true;
+    }
+    if (amount > can_contrib_total)
+    {
+      fail_msg_writer() << tr("You may only contribute up to ") << print_money(can_contrib_total) << tr(" more loki to this service node");
+      return true;
+    }
+    if (amount < must_contrib_total)
```

Comment:
This feels like it should just be a printed message, not a fail case. Use case, person has 2 wallets and wants to contribute funds from each, they shouldn't be required to consolidate their funds.

Reply:
Yes, you're right. Good idea.